doc/macsec/MACsec_hld.md

```diff
@@ -0,0 +1,613 @@
+    - [3.3.1 MACsec Port Table](#331-macsec-port-table)
+    - [3.3.2 MACsec Egress SC Table](#332-macsec-egress-sc-table)
+    - [3.3.3 MACsec Ingress SC Table](#333-macsec-ingress-sc-table)
+    - [3.3.4 MACsec Egress SA Table](#334-macsec-egress-sa-table)
+    - [3.3.5 MACsec Ingress SA Table](#335-macsec-ingress-sa-table)
+  - [3.4 Software Modules](#34-software-modules)
+    - [3.4.1 MACsec Mgr](#341-macsec-mgr)
+      - [3.4.1.1 Primary/Fallback decision](#3411-primaryfallback-decision)
+      - [3.4.1.2 wpa_cli parameters list](#3412-wpa_cli-parameters-list)
+      - [3.4.1.3 Enable MACsec](#3413-enable-macsec)
+      - [3.4.1.4 Disable MACsec](#3414-disable-macsec)
+    - [3.4.2 WPA Supplicant](#342-wpa-supplicant)
+      - [3.4.2.1 Extension Parameters](#3421-extension-parameters)
+      - [3.4.2.2 Proactive SAK refresh](#3422-proactive-sak-refresh)
+      - [3.4.2.3 Scalability Evaluation](#3423-scalability-evaluation)
+    - [3.4.3 SONiC MACsec Plugin](#343-sonic-macsec-plugin)
+    - [3.4.4 MACsec Orch](#344-macsec-orch)
+      - [3.4.4.1 Functions](#3441-functions)
+      - [3.4.4.2 Flex Counter Interval](#3442-flex-counter-interval)
+    - [3.4.5 vMACsec SAI](#345-vmacsec-sai)
+- [4 Flow](#4-flow)
+  - [4.1 Init Port](#41-init-port)
+  - [4.2 Init MACsec](#42-init-macsec)
+  - [4.3 Create and Enable SA](#43-create-and-enable-sa)
+
+## About this Manual
+
+This document provides general information about the MACsec feature implementation in SONiC.
+
+## Abbreviation
+
+| Abbreviation | Description                              |
+| ------------ | ---------------------------------------- |
+| CA           | Secure Connectivity Association          |
+| CAK          | Secure Connectivity Association Key      |
+| CAN          | Secure Connectivity Association Key Name |
+| SA           | Secure Association                       |
+| SAK          | Secure Association Key                   |
+| SC           | Secure Channel                           |
+| SCI          | Secure Channel Identifier                |
+
+## 1 Requirements Overview
+
+### 1.1 Functional requirements
+
+This section describes the SONiC requirements for MACsec primary feature.
+
+At a high level the following should be supported:
+
+#### Phase I
+
+- MACsec should be supported on physical port
+- Support Cipher: GCM-AES-128 and GCM-AES-256
+- Secure Association Key(SAK) can be replaced without service outage
+
+#### Phase II
+
+- MACsec can support  Extension packet number(XPN), which means to support Cipher Suites: GCM-AES-XPN-128 and GCM-AES-XPN-256
+- SAK can be refreshed proactively.
+- Primary and Fallback secure Connectivity Association Key can be supported simultaneously.
+- Choose the cipher suite by the wpa_cli
+- A part of parameters of wpa_supplicant can be updated on the fly
+
+#### Phase III
+
+- CLI commands to configure MACsec
+
+## 2 Architecture Design
+
+This chapter shows the MACsec interface stack of virtual switch and real switch.
```

Comment:
Some of the container parts are missing in design. We should provide some details on the following:
1. When will the container be running
2. Is the FEATURE disabled/enabled by default

Reply:
When will the container be running
Add the information at : https://github.com/Azure/SONiC/pull/652/files#diff-600ad1f148d06a6c05f027cd045d6c4fR122
Is the FEATURE disabled/enabled by default
Add the information at : https://github.com/Azure/SONiC/pull/652/files#diff-600ad1f148d06a6c05f027cd045d6c4fR198-R199